src/configure/clients/modaClient.ts

```diff
@@ -0,0 +1,26 @@
+import { IRestResponse, RestClient } from "typed-rest-client";
+import vscodeUri from "vscode-uri";
+import { RepositoryAnalysisRequest } from "../model/models";
+
+export class ModaClient{
+    private restClient: RestClient;
+    private githubPat: string;
+    private pathUrl: string;
+
+    constructor(url: string, githubPat: string) {
+        const u = vscodeUri.parse(url);
+        this.restClient = new RestClient("deploy-to-azure", u.scheme + "://" + u.authority);
```

Comment:
what is the second parameter? is it creating a value of type `ServiceClientCredentials`? if so, why not take this type in the constructor. this way both PEClient and ModaClient will have same constructor signature- - and easier to derive from same interface.

Reply:
no it's not creating value of type ServiceClientCredentials. Second parameter is the base url, in our case "https://aka.ms"